### PR TITLE
enh(mp): Squid SNMP

### PR DESCRIPTION
**Pull request by itoussies**
## Description

Documentation enhancements using the new metrics naming convention and new monitoring procedure.

## Target serie

- [X] 20.04.x
- [X] 20.10.x
- [X] 21.04.x
- [X] 21.10.x (master)

### CODE REVIEW

**Comment by itoussies** [comment]
NOT READY

File: en/integrations/plugin-packs/procedures/applications-squid-snmp.md [hunk 1]
```diff
@@ -3,47 +3,171 @@ id: applications-squid-snmp
...
+| http.errors.count              | errors: %s           | count |
+| http.traffic.in.bitspersecond  | traffic in: %s %s/s  | b/s   |
+| http.traffic.out.bitspersecond | traffic out: %s %s/s | b/s   |
+
+* ICP statistics
+
+| Metric name                   | Description          |
+|:------------------------------|:---------------------|
+| icp.traffic.in.bitspersecond  | traffic in: %s %s/s  |
+| icp.traffic.out.bitspersecond | traffic out: %s %s/s |
```
Comment (thibaults-centreon) [inline en/integrations/plugin-packs/procedures/applications-squid-snmp.md:42]: Missing unit column

File: en/integrations/plugin-packs/procedures/applications-squid-snmp.md [hunk 2]
```diff
@@ -3,47 +3,171 @@ id: applications-squid-snmp
...
-Read [Troubleshooting
-SNMP](https://documentation.centreon.com/docs/centreon-plugins/en/latest/user/guide.html#snmp).
+Expected command output is shown below:
+
+```bash
+OK : Cpu usage: 30 % Memory usage: 265289728 B Number of file descriptors: 45 Number of object stored: 23 | 'cache.cpu.utilization.percentage'=30%;;;0;100 
+'cache.memory.usage.bytes'=265289728B;;;0; 'cache.filedescriptors.count'=45;;;0; 'cache.objects.count'=23;;;0; 
+```
+
+This command would trigger a WARNING alarm if the *cpu* utilizationis reported
```
Comment (thibaults-centreon) [inline en/integrations/plugin-packs/procedures/applications-squid-snmp.md:138]:
```suggestion
This command would trigger a WARNING alarm if the *cpu* utilization is reported 
```

File: en/integrations/plugin-packs/procedures/applications-squid-snmp.md [hunk 3]
```diff
@@ -3,47 +3,171 @@ id: applications-squid-snmp
...
+OK : Cpu usage: 30 % Memory usage: 265289728 B Number of file descriptors: 45 Number of object stored: 23 | 'cache.cpu.utilization.percentage'=30%;;;0;100 
+'cache.memory.usage.bytes'=265289728B;;;0; 'cache.filedescriptors.count'=45;;;0; 'cache.objects.count'=23;;;0; 
+```
+
+This command would trigger a WARNING alarm if the *cpu* utilizationis reported 
+as over 80% (`--warning-cpu='80'`) and a CRITICAL alarm if over 90% 
+(`--critical-cpu='90'`).
+
+All available options for a given mode can be displayed by adding the 
+`--help` parameter to thecommand:
```
Comment (thibaults-centreon) [inline en/integrations/plugin-packs/procedures/applications-squid-snmp.md:143]:
```suggestion
`--help` parameter to the command:
```

File: en/integrations/plugin-packs/procedures/applications-squid-snmp.md [hunk 4]
```diff
@@ -3,47 +3,171 @@ id: applications-squid-snmp
...
+
+```bash
+/usr/lib/centreon/plugins//centreon_squid_snmp.pl  \
+    --plugin=apps::squid::snmp::plugin  \
+    --mode=cache-usage  \
+    --help
+```
+
+All available options for a given mode can be displayed by adding the 
+`--list-mode` parameter to thecommand:
```
Comment (thibaults-centreon) [inline en/integrations/plugin-packs/procedures/applications-squid-snmp.md:153]:
```suggestion
`--list-mode` parameter to the command:
```

File: en/integrations/plugin-packs/procedures/applications-squid-snmp.md [hunk 5]
```diff
@@ -3,47 +3,171 @@ id: applications-squid-snmp
...
-| Host name               | *Name of the host*         |
-| Alias                   | *Host description*         |
-| IP                      | *Host IP Address*          |
-| Monitored from          | *Monitoring Poller to use* |
-| Host Multiple Templates | App-Squid-SNMP-custom      |
+#### UNKNOWN: SNMP GET Request : Cant get a single value
 
-Click on the *Save* button.
+This message generally means that SNMP privileges are not wide enough for
+the mode/plugin to work properly.
```
Comment (thibaults-centreon) [inline en/integrations/plugin-packs/procedures/applications-squid-snmp.md:173]:
```suggestion
the mode/Plugin to work properly.
```

File: fr/integrations/plugin-packs/procedures/applications-squid-snmp.md [hunk 6]
```diff
@@ -3,47 +3,167 @@ id: applications-squid-snmp
...
+| http.errors.count              | errors: %s           | count |
+| http.traffic.in.bitspersecond  | traffic in: %s %s/s  | b/s   |
+| http.traffic.out.bitspersecond | traffic out: %s %s/s | b/s   |
+
+* ICP statistics
+
+| Metric name                   | Description          |
+|:------------------------------|:---------------------|
+| icp.traffic.in.bitspersecond  | traffic in: %s %s/s  |
+| icp.traffic.out.bitspersecond | traffic out: %s %s/s |
```
Comment (thibaults-centreon) [inline fr/integrations/plugin-packs/procedures/applications-squid-snmp.md:42]: missing unit column

File: fr/integrations/plugin-packs/procedures/applications-squid-snmp.md [hunk 7]
```diff
@@ -3,47 +3,167 @@ id: applications-squid-snmp
...
+CRITICAL au-delà de 90%.
+
+```bash
+/usr/lib/centreon/plugins//centreon_squid_snmp.pl  \
+    --plugin=apps::squid::snmp::plugin  \
+    --mode=cache-usage  \
+    --help
+```
+
+Tous les modes disponibles peuvent être affichés en ajoute le paramètre
```
Comment (thibaults-centreon) [inline fr/integrations/plugin-packs/procedures/applications-squid-snmp.md:149]:
```suggestion
Tous les modes disponibles peuvent être affichés en ajoutant le paramètre 
```

File: fr/integrations/plugin-packs/procedures/applications-squid-snmp.md [hunk 8]
```diff
@@ -3,47 +3,167 @@ id: applications-squid-snmp
...
+### Diagnostic des erreurs communes
 
-### Create a host using the appropriate template
+#### UNKNOWN: SNMP GET Request : Timeout
 
-Go to *Configuration \> Hosts* and click *Add*. Then, fill the form as shown by
-the following table:
+Si vous obtenez ce message, cela signifie le collecteur Centreon ne parvientpas
+à contacter le serveur sur le port 161 (firewallou autre équipement en coupure)
+ou que la communauté SNMP configurée n'est pascorrecte.
```
Comment (thibaults-centreon) [inline fr/integrations/plugin-packs/procedures/applications-squid-snmp.md:164]:
```suggestion
Si vous obtenez ce message, cela signifie le collecteur Centreon ne parvient pas
à contacter le serveur sur le port 161 (firewall ou autre équipement en coupure)
ou que la communauté SNMP configurée n'est pas correcte.
```

File: fr/integrations/plugin-packs/procedures/applications-squid-snmp.md [hunk 9]
```diff
@@ -3,47 +3,167 @@ id: applications-squid-snmp
...
-| Host name               | *Name of the host*         |
-| Alias                   | *Host description*         |
-| IP                      | *Host IP Address*          |
-| Monitored from          | *Monitoring Poller to use* |
-| Host Multiple Templates | App-Squid-SNMP-custom      |
+#### UNKNOWN: SNMP GET Request : Cant get a single value
 
-Click on the *Save* button.
+Les autorisations données à l'utilisateur en SNMP sont trop restreintes pour
+faire fonctionner le mode/plugin.
```
Comment (thibaults-centreon) [inline fr/integrations/plugin-packs/procedures/applications-squid-snmp.md:169]:
```suggestion
faire fonctionner le mode/Plugin.
```

**Review by thibaults-centreon — CHANGES_REQUESTED** [review]

**Review by thibaults-centreon — APPROVED** [review]

**Pull request merged**